src/v/raft/replicate_batcher.cc

```diff
@@ -110,7 +111,8 @@ ss::future<replicate_batcher::item_ptr> replicate_batcher::do_cache(
   std::optional<model::term_id> expected_term,
   model::record_batch_reader r,
   replicate_options opts) {
-    auto batches = co_await model::consume_reader_to_memory(
+    auto batches = co_await model::consume_reader_to_fragmented_memory<
+      std::dynamic_extent>(
```

Comment:
nit: this reads poorly imo, can we alias this version to `consume_reader_to_chunked_vector`?

Reply:
sure, will do it